### ISSUE TITLE
Cannot retrieve userinfo with content-type `application/x-www-form-urlencoded; charset=UTF-8`

### ISSUE THREAD

**Issue by atosscsd-bb** [keycloak/keycloak#19494]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

oidc

### Describe the bug

When retrieving the userinfo through an HTTP POST request and content-type `application/x-www-form-urlencoded; charset=UTF-8`, keycloak is replying with an _invalid_token_ error.

### Version

21.0.2

### Expected behavior

Keycloak should return the userinfo correlated to passed token.

### Actual behavior

Keycloak isn't return the userinfo, but reporting an `invalid_token` error.

### How to Reproduce?

Obtain an access token and query the corresponding userinfo:
```
curl -L -X POST 'http://localhost:8080/realms/ATC/protocol/openid-connect/userinfo' \
 -H 'Content-Type: application/x-www-form-urlencoded; charset=UTF-8' \
 --data-urlencode 'access_token=<TOKEN>'
```

Keycloak is responding with http status 401 and following log entry:
```
WARN  [org.keycloak.events] (executor-thread-5) type=USER_INFO_REQUEST_ERROR, realmId=<REALMID>, clientId=null, userId=null, ipAddress=10.0.7.1, error=invalid_token, auth_method=validate_access_token
```

### Anything else?

As the charset is a valid part of the [content-type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type) it should be accepted. Previous versions of keycloak (18?) have accepted this type. 

The issue was discovered through [nimbus](https://www.javadoc.io/doc/com.nimbusds/oauth2-oidc-sdk/latest/com/nimbusds/openid/connect/sdk/UserInfoRequest.html). This library is querying userinfo with said content type.

**Comment by mposolda** [comment]
I am closing as this should be already fixed in latest Keycloak main and will be fixed in next Keycloak release (probably 21.1.0).

If you want to doublecheck, you can try latest Keycloak nightly release https://github.com/keycloak/keycloak/releases/tag/nightly or build from sources. Please re-open just if you're able to reproduce with latest nightly or latest sources.

**Comment by atosscsd-bb** [comment]
@mposolda  Running my scenario with docker image `quay.io/keycloak/keycloak:nightly` is showing the same error.

I'm not sure which version of keycloak this image bundles, the container is reporting version 999.0:
```
[io.quarkus] (main) Keycloak 999.0.0-SNAPSHOT on JVM (powered by Quarkus 2.13.7.Final)
```
